lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableDefaultRemoteManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Util;
+
+namespace Org.Apache.REEF.Wake.Remote.Impl
+{
+    /// <summary>
+    /// Manages incoming and outgoing messages between remote hosts.
+    /// </summary>
+    /// <typeparam name="T">Message type T. It is assumed to be IWritable</typeparam>
+    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    public class WritableDefaultRemoteManager<T> : IRemoteManager<T> where T : IWritable
```

Comment:
In [REEF-290](https://issues.apache.org/jira/browse/REEF-290), we are discussing ways to get rid of constructor calls for these classes. In [REEF-283](https://issues.apache.org/jira/browse/REEF-283) we added an injectable factory for these to make sure we can configure this part of the stack with Tang. It might be worthwhile to look into these. Otherwise, the work to merge this PR as well as the upcoming ones for [REEF-273](https://issues.apache.org/jira/browse/REEF-273) and [REEF-274](https://issues.apache.org/jira/browse/REEF-274) will be tricky and laborious.